nelox/interpreter.py

```diff
@@ -0,0 +1,223 @@
+from nelox.Expr import Literal, Variable, List
+
+
+class Environment:
+    def __init__(self, parent=None):
+        self.values = {}
+        self.parent = parent
+
+    def get(self, name):
+        if name in self.values:
+            return self.values[name]
+        elif self.parent:
+            return self.parent.get(name)
+        else:
+            raise RuntimeError(f"Undefined variable '{name}'")
```

Comment:
As defined here, this else ends up in every recursive call, but it's actually not relevant in most of them.  I would recommend splitting "find what environment name is defined in" from "get/set that name".